modules/extension/xsd/xsd-core/src/main/java/org/geotools/xml/impl/ParseExecutor.java

```diff
@@ -328,11 +334,16 @@ public XSDNamedComponent getDeclaration() {
             // for mixed
             if (instance.getTypeDefinition() instanceof XSDComplexTypeDefinition
                     && ((XSDComplexTypeDefinition) instance.getTypeDefinition()).isMixed()) {
-                //collape the text
-                text = Whitespace.COLLAPSE.preparse(text);
+                //Collapse the text
+                //Seems to be an issue for transaction updates - See GEOS-8227
+                if(!parser.isCDATA()) {
+                    text = Whitespace.COLLAPSE.preparse(text);
+                }
             }
         }
-
+        if(!parser.isCDATA()) {
```

Comment:
Formatting ifs (I do that too, @dromagnoli  catches me every time): ``if (!parser.isCDATA())``